RoaringBitmap/src/main/java/org/roaringbitmap/longlong/Roaring64NavigableMap.java

```diff
@@ -692,6 +692,53 @@ private int highestHigh() {
     return RoaringIntPacking.highestHigh(signedLongs);
   }
 
+  /**
+   * to be used with naivelazyor
+   */
+  public void repairAfterLazy() {
+    for (Entry<Integer, BitmapDataProvider> e2 : highToBitmap.entrySet()) {
+      BitmapDataProvider lowBitmap2 = e2.getValue();
+      if (lowBitmap2 instanceof RoaringBitmap) {
+        ((RoaringBitmap) lowBitmap2).repairAfterLazy();
+      } else {
+        throw new UnsupportedOperationException(
+                ".repairAfterLazy is not supported for " + lowBitmap2.getClass());
+      }
+    }
+  }
+  /**
+   * In-place bitwise OR (union) operation without maintaining cardinality.
+   * Don't forget to call repairAfterLazy() afterward. The current bitmap is modified.
+   *
+   * @param x2 other bitmap
+   */
+  public void naivelazyor(final Roaring64NavigableMap x2) {
+    boolean firstBucket = true;
+    for (Entry<Integer, BitmapDataProvider> e2 : x2.highToBitmap.entrySet()) {
+      // Keep object to prevent auto-boxing
+      Integer high = e2.getKey();
+
+      BitmapDataProvider lowBitmap1 = this.highToBitmap.get(high);
+
+      BitmapDataProvider lowBitmap2 = e2.getValue();
+
+      if ((lowBitmap1 == null || lowBitmap1 instanceof RoaringBitmap)
+              && lowBitmap2 instanceof RoaringBitmap) {
+        if (lowBitmap1 == null) {
+          // Clone to prevent future modification of this modifying the input Bitmap
+          RoaringBitmap lowBitmap2Clone = ((RoaringBitmap) lowBitmap2).clone();
+
+          pushBitmapForHigh(high, lowBitmap2Clone);
+        } else {
+          ((RoaringBitmap) lowBitmap1).naivelazyor((RoaringBitmap) lowBitmap2);
+        }
+      } else {
+        throw new UnsupportedOperationException(
+                ".naivelazyor is not between " + this.getClass() + " and " + lowBitmap2.getClass());
```

Comment:
A pity we do not report the class of the nullable lowBitmap1. I would prefer 2 throws, one if lowBitmap1 is null, and another if not null.

 ```
     if (!(lowBitmap2 instanceof RoaringBitmap)){
        throw ...
     } else if (lowBitmap1 != null && !(lowBitmap1 instanceof RoaringBitmap)) {
        throw ...
    } else {
             ...
    }
```

Just a matter of style anyway.

Reply:
Given the code over RoaringBitmap and MutableRoaringBitmap, I expected even more relevant to process. Hence I took time to code the suggested alternative. What do you think @huangfeng1993 ? 

```
      lowBitmap1 == null specifically, to have finer exceptions, and easier code to read:

      if (lowBitmap1 == null) {
        // Clone to prevent future modification of this modifying the input Bitmap
        BitmapDataProvider lowBitmap2Clone;
        if (lowBitmap2 instanceof RoaringBitmap) {
          lowBitmap2Clone = ((RoaringBitmap) lowBitmap2).clone();
        } else if (lowBitmap2 instanceof MutableRoaringBitmap) {
          lowBitmap2Clone = ((MutableRoaringBitmap) lowBitmap2).clone();
        } else {
          throw new UnsupportedOperationException(
                  ".naivelazyor is not between " + this.getClass() + " and " + lowBitmap2.getClass());
        }

        pushBitmapForHigh(high, lowBitmap2Clone);
      } else if (lowBitmap1 instanceof RoaringBitmap && lowBitmap2 instanceof RoaringBitmap) {
        RoaringBitmapPrivate.naivelazyor((RoaringBitmap) lowBitmap1, (RoaringBitmap) lowBitmap2);
      } else if (lowBitmap1 instanceof MutableRoaringBitmap && lowBitmap2 instanceof MutableRoaringBitmap) {
        MutableRoaringBitmapPrivate.naivelazyor((MutableRoaringBitmap) lowBitmap1, (MutableRoaringBitmap) lowBitmap2);
      } else {
        throw new UnsupportedOperationException(
                ".naivelazyor is not between " + lowBitmap1.getClass() + " and " + lowBitmap2.getClass());
      }
```

Reply:
It's fine.